src/util/getTopologicalOrder.ts

```diff
@@ -0,0 +1,56 @@
+export interface Edge {
+	origin: string;
+	target: string;
+}
+
+// Uses the Khan Algorithm loosely
+export default function getTopologicalOrder(inputGraph: ReadonlySet<Edge>): string[] | Set<Edge> {
+	const graph: Set<Edge> = new Set(inputGraph);
+
+	const order: string[] = [];
+	const vertexesWithNoInEdge: Set<string> = getVertexesWithNoInEdge(graph);
+
+	// These are to avoid issues described in following thread by externalizing deletions (from the loop):
+	// https://github.com/diproche/webinterface/pull/28/files/4647fd1d5b3e9050a532c5c24f98d14643f1661e#r287705437
+	const edgesToRemove: Set<Edge> = new Set();
+	const vertexesToRemove: Set<string> = new Set();
+
+ while (vertexesWithNoInEdge.size !== 0) {
+		vertexesWithNoInEdge.forEach((vertex: string) => {
+			vertexesToRemove.add(vertex);
+			order.push(vertex);
+			graph.forEach((edge: Edge) => {
+					if (edge.origin === vertex) { edgesToRemove.add(edge); }
+					vertexesWithNoInEdge.add(edge.target);
```

Comment:
You are currently iterating over `vertexesWithNoInEdge` (line 19).

Reply:
I think in this case it's fine, because I am only adding values(?). The issue with Strings and maybe Sets seemed to have been, that you possible jumped a value.

Reply:
Otherwise I can of course externalize the addition too, I just wanted to keep readability as good as possible.